source/grid/tria_accessor.cc

```diff
@@ -1308,11 +1311,92 @@ namespace
     if (std::abs((v03 * normal) * (v03 * normal) /
                  ((v03 * v03) * (v01 * v01) * (v02 * v02))) >= 1e-24)
       {
-        Assert(
-          false,
-          ExcMessage(
-            "Computing the measure of a nonplanar face is not implemented!"));
-        return std::numeric_limits<double>::quiet_NaN();
+        // If the vectors are non planar we integrate the norm of the normal
+        // vector using a numerical Gauss scheme of order 4.
```

Comment:
This is a bit too abstract even for me. Can you write out in formulas what we are integrating here?

Reply:
Sure, I've added the description of the procedure.